src/Microsoft.Android.Build.BaseTasks/AsyncTask.cs

```diff
@@ -0,0 +1,431 @@
+
+		public void LogMessage(string message, params object[] messageArgs) => LogMessage(string.Format(message, messageArgs));
+
+		public void LogDebugMessage(string message) => LogMessage(message, importance: MessageImportance.Low);
+
+		public void LogDebugMessage(string message, params object[] messageArgs) => LogMessage(string.Format(message, messageArgs), importance: MessageImportance.Low);
+
+		public void LogMessage(string message, MessageImportance importance = MessageImportance.Normal)
+		{
+			if (uiThreadId == Thread.CurrentThread.ManagedThreadId)
+			{
+#pragma warning disable 618
+				Log.LogMessage(importance, message);
+				return;
+#pragma warning restore 618
+			}
+
+			var data = new BuildMessageEventArgs(
+					message: message,
+					helpKeyword: null,
+					senderName: null,
+					importance: importance
+			);
+			EnqueueMessage(logMessageQueue, data, logDataAvailable);
+		}
+
+		public void LogError(string message) => LogCodedError(code: null, message: message, file: null, lineNumber: 0);
+
+		public void LogError(string message, params object[] messageArgs) => LogCodedError(code: null, message: string.Format(message, messageArgs));
+
+		public void LogErrorFromException(Exception exception)
+		{
+			if (uiThreadId == Thread.CurrentThread.ManagedThreadId)
+			{
+#pragma warning disable 618
+				Log.LogErrorFromException(exception);
+				return;
+#pragma warning restore 618
+			}
+
+			StackFrame exceptionFrame = null;
+			try
+			{
+				exceptionFrame = new StackTrace(exception, true)?.GetFrames()?.FirstOrDefault();
+			}
+			catch { }
+
+			lock (errorMessageQueue.SyncRoot)
+			{
+				errorMessageQueue.Enqueue(new BuildErrorEventArgs(
+					subcategory: null,
+					code: null,
+					file: exceptionFrame?.GetFileName(),
+					lineNumber: exceptionFrame?.GetFileLineNumber() ?? 0,
+					columnNumber: exceptionFrame?.GetFileColumnNumber() ?? 0,
+					endLineNumber: 0,
+					endColumnNumber: 0,
+					message: exception.Message,
+					helpKeyword: null,
+					senderName: null
+				));
+				lock (eventlock)
+				{
+					if (isRunning)
+						errorDataAvailable.Set();
+				}
+			}
+		}
+
+		public void LogCodedError(string code, string message) => LogCodedError(code: code, message: message, file: null, lineNumber: 0);
+
+		public void LogCodedError(string code, string message, params object[] messageArgs) => LogCodedError(code: code, message: string.Format(message, messageArgs), file: null, lineNumber: 0);
+
+		public void LogCodedError(string code, string file, int lineNumber, string message, params object[] messageArgs) => LogCodedError(code: code, message: string.Format(message, messageArgs), file: file, lineNumber: lineNumber);
+
+		public void LogCodedError(string code, string message, string file, int lineNumber)
+		{
+			if (uiThreadId == Thread.CurrentThread.ManagedThreadId)
+			{
+#pragma warning disable 618
+				Log.LogError(
+					subcategory: null,
```

Comment:
Indentation is inconsistent.  *My* preference is "parameters indented *two* tab stops*, which is done ~15 lines below!  *Here* parameters are indented *one* tab stop, resulting in "odd" alignment wherein *this* invocation, nested in an `if`, has the same indentation as the subsequent invocation, which *isn't* nested in an `if`!